### ISSUE TITLE
Promising resultas for spinal canal stenosis using axial slices and segmentations

### ISSUE THREAD

**Issue by abelsalm** [ivadomed/lumbar-classification-rsna-challenge-2024#34]
I tried training a ResNet with two channels to use the axial patches of the spinal canal and their spinal canal segmentations at once.
I'll try more models, updating also hyperparameters, and comparing the results with only using the images and not the segmentations, but still this first training led to very promising resultas for this pathology : 
![image](https://github.com/user-attachments/assets/3f4292ee-e264-4f75-a9e7-8fb6c81f5b25)
It is good since the mean loss on all of the pathologies of the best competitor is actually 0.33. 

The code I used is on the branch abels

We could try something similar for the subarticular stenosis, I hope resultas will be good.

Also I'm trying the same model (with less layers) to train it 5 times on five the five different levels to see if we have a better accuracy.

**Comment by abelsalm** [comment]
@jcohenadad If you want to have a look!

**Comment by jcohenadad** [comment]
Very cool results! Just to clarify, this is using the [1,2,4] loss, right? (ie: the same that will be used by the challenge for evaluation)

**Comment by abelsalm** [comment]
> Very cool results! Just to clarify, this is using the [1,2,4] loss, right? (ie: the same that will be used by the challenge for evaluation)

Yes indeed!

**Comment by abelsalm** [comment]
I trained more models, all of the results are on https://github.com/ivadomed/lumbar-classification-rsna-challenge-2024/tree/abels/code/models_per_level

We can compare the training done using only the image patches, with a unique channel model, and the one done for a 2 channels model, using both image and seg, showing that adding this second channels really makes a difference.

Not considering overfitting, we can observe the loss is slightely higher here : (with images only)
![image](https://github.com/user-attachments/assets/f0375266-859f-4b4b-873e-6b2d2861e9cf)

Than here : (images +segmentations of spinal canal)
![image](https://github.com/user-attachments/assets/2e715082-88a5-4b2e-9876-148945bf59ae)

**Comment by abelsalm** [comment]
Last training, saving the best model.
It had 0.293 for the best validation loss and then 0.30 for the test
![image](https://github.com/user-attachments/assets/ae6f5645-ab1b-45c8-957a-8f7f8e8e3837)
